### PR TITLE
CLN: remove _recast_datetimelike_result

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -1283,7 +1282,7 @@ def _wrap_applied_output_series(
         # as we are stacking can easily have object dtypes here
         so = self._selected_obj
         if so.ndim == 2 and so.dtypes.apply(needs_i8_conversion).any():
-            result = _recast_datetimelike_result(result)
+            result = result._convert(datetime=True)
```
Comment (jreback) [inline pandas/core/groupby/generic.py:1285]: ```.apply(to_datetime)``` ?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/groupby/generic.py:1285]: we only want to do it on object-dtype columns which is what _convert does

Comment (jreback) [reply pandas/core/groupby/generic.py:1285]: kk cool, then this is fine. ping on green.

**Review by jreback — APPROVED** [review]

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/groupby/generic.py:1285]: ping

**Pull request merged**